aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -178,7 +208,13 @@ private void newStream() throws IOException {
     createStagingDirectoryIfNotExists();
     currentStagingFile = File.createTempFile("s3fileio-", ".tmp", stagingDirectory);
     currentStagingFile.deleteOnExit();
-    stagingFiles.add(currentStagingFile);
+    try {
+      currentPartMessageDigest = isEtagCheckEnabled ? MessageDigest.getInstance(digestAlgorithm) : null;
+    } catch (NoSuchAlgorithmException e) {
+      throw new IOException(e);
```

Comment:
Why does this throw `IOException` with no context? I think there should be context at a minimum. I also suspect that this is throw `IOException` because that's an allowed checked exception?

Reply:
We don't want to change the signature, but we also don't want to throw some exception just because it was convenient. The best thing to do is to wrap in RuntimeException, which avoids throwing a misleading exception class.

Reply:
Sure, changed this to RuntimeException.